app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java

```diff
@@ -0,0 +1,137 @@
+package com.eveningoutpost.dexdrip.Services;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.eveningoutpost.dexdrip.Models.UserError.Log;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import retrofit.Call;
+import retrofit.Response;
+import retrofit.http.GET;
+import retrofit.http.Query;
+import retrofit.Retrofit;
+import retrofit.GsonConverterFactory;
+
+import com.squareup.okhttp.logging.HttpLoggingInterceptor;
+import com.squareup.okhttp.OkHttpClient;
+
+
+public class NsRestApiReader {
+
+	class NightscoutBg {
+	    double xDrip_raw; // raw_data
+	    double xDrip_filtered; // filtered_data;
+	    Long date; // timestamp
+	    double sgv; // calculated_bg
+	}
+
+	class NightscoutMbg {
+	    Long date; // timestamp
+	    double mbg; // calculated_bg
+	    double xDrip_slope;
+	    double xDrip_intercept;
+	    double xDrip_estimate_raw_at_time_of_calibration;
+	    double xDrip_slope_confidence;
+	    double xDrip_sensor_confidence;
+	    double xDrip_raw_timestamp;
```

Comment:
Assuming this is a record to store calibrations in: This is what xDrip later would want. This mapping then should be done by the receiver-Function in xDrip that gets the data from the Service handling this library.

Nightscout doesn't usually have these fields in a calibration. A calibration looks like this: {"_id":"56641b8adbb114d587f57ab6","device":"xDrip-DexcomShare","type":"cal","date":1449401035638,"dateString":"2015-12-06T12:23:55.638+0100","slope":745,"intercept":27407,"scale":1}

This looks like the combination of a meter entry, a calibration entry and some more. A combination that is not needed in "Viewer" mode in xDrip anyways (except for the calibration graph maybe).